kubectl-plugin/pkg/cmd/submit/submit.go

```diff
@@ -0,0 +1,537 @@
+		return fmt.Errorf("no context is currently set, use %q to select a new one", "kubectl config use-context <context>")
+	}
+
+	if len(options.runtimeEnv) > 0 {
+		info, err := os.Stat(filepath.Clean(options.runtimeEnv))
+		if os.IsNotExist(err) {
+			return fmt.Errorf("Runtime Env file does not exist. Failed with: %w", err)
+		} else if err != nil {
+			return fmt.Errorf("Error occurred when checking runtime env file: %w", err)
+		} else if !info.Mode().IsRegular() {
+			return fmt.Errorf("Filename given is not a regular file. Failed with: %w", err)
+		}
+
+		hasWorkingDir, err := runtimeEnvHasWorkingDir(filepath.Clean(options.runtimeEnv))
+		if err != nil {
+			return fmt.Errorf("Error while checking runtime env: %w", err)
+		}
+		if hasWorkingDir && options.fileName == "" {
+			return fmt.Errorf("Ray Job file -f/--filename is required when using env variable `working_dir`")
+		}
+	}
+
+	if len(options.workingDir) > 0 && options.fileName == "" {
+		return fmt.Errorf("Ray Job file -f/--filename is required when using --working-dir")
+	}
+
+	if len(options.fileName) > 0 {
+		info, err := os.Stat(filepath.Clean(options.fileName))
+		if os.IsNotExist(err) {
+			return fmt.Errorf("Ray Job file does not exist. Failed with: %w", err)
+		} else if err != nil {
+			return fmt.Errorf("Error occurred when checking ray job file: %w", err)
+		} else if !info.Mode().IsRegular() {
+			return fmt.Errorf("Filename given is not a regular file. Failed with: %w", err)
+		}
+
+		options.RayJob, err = decodeRayJobYaml(options.fileName)
+		if err != nil {
+			return fmt.Errorf("Failed to decode RayJob Yaml: %w", err)
+		}
+
+		submissionMode, ok := options.RayJob.Object["spec"].(map[string]interface{})["submissionMode"]
+		if !ok {
+			return fmt.Errorf("RayJob does not have `submissionMode` field set")
+		}
+		if submissionMode != nil {
+			// Currently using string since latest release does not have `rayv1api.UserMode` yet
+			if submissionMode != "UserMode" {
+				return fmt.Errorf("Submission mode of the Ray Job is not supported")
+			}
+		} else {
+			return fmt.Errorf("Submission mode must be set to rayv1api.UserMode or `UserMode`")
+		}
+	}
+
+	return nil
+}
+
+func (options *SubmitJobOptions) Run(ctx context.Context, factory cmdutil.Factory) error {
+	k8sClients, err := client.NewClient(factory)
+	if err != nil {
+		return fmt.Errorf("failed to initialize clientset: %w", err)
+	}
+
+	if options.RayJob != nil {
+		// createdRayJob, err = k8sClients.CreateRayCustomResource(ctx, util.RayJob, options.configFlags.Namespace, unstructuredRayjob)
+		options.RayJob, err = k8sClients.DynamicClient().Resource(util.RayJobGVR).Namespace(*options.configFlags.Namespace).Create(ctx, options.RayJob, v1.CreateOptions{})
+		if err != nil {
+			return fmt.Errorf("Error when creating RayJob CR: %w", err)
+		}
+		fmt.Printf("Submitted RayJob %s.\n", options.RayJob.GetName())
+	}
+
+	if options.RayJob != nil && len(options.RayJob.GetName()) > 0 {
```

Comment:
The first condition can be removed and this if block can be moved to the if block above

Reply:
Yep! thanks. Done!